stdlib/stdlib-external.mk

```diff
@@ -0,0 +1,29 @@
+# Define a set of targets for downloading, caching, and building the source of
+# a stdlib library. The commit and git branch should be defined in a file
+# $stdlib_name.version in the current directory. See the git-external macro for
+# additional documentation.
+#
+# Parameters to the stdlib-external macro:
+#
+#   $1 = stdlib_name
+#   $2 = var_prefix (by convention, use upper cased stdlib_name)
+
+include $(JULIAHOME)/deps/tools/git-external.mk
+
+define stdlib-external
+
+$2_GIT_URL := git://github.com/JuliaLang/$1.jl.git
```

Comment:
perhaps the `JuliaLang` part should also be a parameter?

Reply:
I was just about to comment that actually, e.g. I created https://github.com/JuliaStdlibs some time ago, but there seemed to be more interest in keeping the stdlibs under the `JuliaLang` organization.

Reply:
I think that ideally we would just define an URL_BASE or similar; people might want to use this to build their own weird Julia versions that use packages hosted on internal git repositories as part of the stdlib.

I think we should just require users to define a `$(2)_GIT_URL` before calling `stdlib-external`.